cli/command/image/build.go

```diff
@@ -469,3 +471,34 @@ func replaceDockerfileTarWrapper(ctx context.Context, inputTarStream io.ReadClos
 
 	return pipeReader
 }
+
+func createBuildArgs(cfg *configfile.ConfigFile, host string, o opts.ListOpts) map[string]*string {
```

Comment:
There's a lot of duplication between this and `parseProxyConfig()`. I think that should be easy to fix.

If you move `createBuildArgs()` to `cli/config/configfile` and make it `ParseProxyConfig()`,  you could call it directly from this package, and in `container/run.go` you could call it and perform the last step of converting to a ListOpts again.

A test case for the new function would be great as well.